pkg/console/subresource/consoleserver/config_builder.go

```diff
@@ -601,3 +602,43 @@ func (b *ConsoleServerCLIConfigBuilder) proxy() Proxy {
 func (b *ConsoleServerCLIConfigBuilder) Telemetry() map[string]string {
 	return b.telemetry
 }
+
+// buildCapabilities will configure the capabilities - use user-defined capabilities if provided, otherwise intelligent defaults
+func (b *ConsoleServerCLIConfigBuilder) buildCapabilities() []operatorv1.Capability {
+	// Get the default capabilities / user-defined capabilities
+	capabilities := b.capabilities
+
+	// No user-defined capabilities, apply default capabilities based on cluster architecture
+	// Default state: Both capabilities are ENABLED (cluster default)
+	// Only disable LightspeedButton if cluster is not homogeneous AMD64
+
+	// Check if cluster is homogeneous AMD64
+	// Only enable Lightspeed if ALL nodes are AMD64 (homogeneous cluster)
+	// nodeArchitectures contains deduplicated architectures from all nodes
+	var isHomogeneousAMD64 bool
+	if len(b.nodeArchitectures) == 1 {
+		isHomogeneousAMD64 = b.nodeArchitectures[0] == "amd64"
+	} else {
+		// Mixed architecture cluster - disable Lightspeed
+		isHomogeneousAMD64 = false
+	}
+
+	// Configure LightspeedButton based on architecture
+	lightspeedState := operatorv1.CapabilityEnabled // Default to enabled
+	if !isHomogeneousAMD64 {
+		lightspeedState = operatorv1.CapabilityDisabled
+		klog.V(4).Infof("disabling LightspeedButton capability for heterogenous cluster, architectures: %v", b.nodeArchitectures)
+	} else {
+		klog.V(4).Infof("enabling LightspeedButton capability for homogeneous AMD64 cluster")
+	}
+
+	if len(capabilities) > 0 {
```

Comment:
I kept the condition check here because, in the test suite, we don't set nodeArchitecture, nor do we provide a default for the capability. As you mentioned, the default capabilities are defined in the YAML file [here](https://github.com/openshift/console-operator/blob/main/manifests/01-operator-config.yaml#L18). Since the test environment doesn't have access to this YAML, some tests may fail due to the lack of these default values.

Please let me know if you have any suggestions for a better approach to resolve this issue. Thanks!

Reply:
I think its ok.
But just to be sure I would rather travers the array and find the appropriate lightspeed item, set it and print the value for the current item so we can drop the line below with the logging